scalafmt-core/native/src/main/scala/org/scalafmt/internal/RegexCompat.scala

```diff
@@ -0,0 +1,232 @@
+        isEscaped = false
+      case '[' if !isEscaped =>
+        sb.append('[')
+        isInClass = true
+      case ']' if !isEscaped =>
+        sb.append(']')
+        isInClass = false
+      case other =>
+        if (isEscaped) {
+          isEscaped = false
+          sb.append('\\')
+        }
+        sb.append(other)
+    }
+    sb.toString()
+  }
+
+  @inline
+  val trailingSpace = Pattern
+    .compile(RegexCompat.fixHorizontalSpaceInRegex("\\h+$"), Pattern.MULTILINE)
+
+  // "slc" stands for single-line comment
+  @inline
+  val slcLine = Pattern
+    .compile(RegexCompat.fixHorizontalSpaceInRegex("^/\\/\\/*\\h*(.*?)\\h*$"))
+
+  @inline
+  val slcDelim = Pattern.compile(RegexCompat.fixHorizontalSpaceInRegex("\\h+"))
+
+  // "mlc" stands for multi-line comment
+  @inline
+  val mlcHeader = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex("^/\\*\\h*(?:\n\\h*[*]*\\h*)?"),
+  )
+
+  @inline
+  val mlcLineDelim = Pattern
+    .compile(RegexCompat.fixHorizontalSpaceInRegex("\\h*\n\\h*[*]*\\h*"))
+
+  @inline
+  val mlcParagraphEnd = Pattern.compile("[.:!?=]$")
+
+  @inline
+  val mlcParagraphBeg = Pattern.compile("^(?:[-*@=]|\\d+[.:])")
+
+  @inline
+  val leadingAsteriskSpace = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex("\n\\h*[*][^*]"),
+    Pattern.MULTILINE,
+  )
+
+  @inline
+  val docstringLine = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex("^(?:\\h*\\*)?(\\h*)(.*?)\\h*$"),
+    Pattern.MULTILINE,
+  )
+
+  @inline
+  val emptyLines = RegexCompat.fixHorizontalSpaceInRegex("\\h*(\n\\h*\\*?\\h*)*")
+
+  @inline
+  val emptyDocstring = Pattern.compile(s"^/\\*\\*$emptyLines\\*/$$")
+
+  @inline
+  val onelineDocstring = {
+    val oneline = RegexCompat
+      .fixHorizontalSpaceInRegex("[^*\n\\h](?:[^\n]*[^\n\\h])?")
+    Pattern.compile(RegexCompat.fixHorizontalSpaceInRegex(
+      s"^/\\*\\*$emptyLines($oneline)$emptyLines\\*/$$",
+    ))
+  }
+
+  @inline
+  val docstringLeadingSpace = Pattern
+    .compile(RegexCompat.fixHorizontalSpaceInRegex("^\\h+"))
+
+  @inline
+  def compileStripMarginPattern(pipe: Char) = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex(s"\n+\\h*?\\$pipe"),
+    Pattern.MULTILINE,
+  )
+
+  @inline
+  def compileStripMarginPatternWithLineContent(pipe: Char) = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex(s"\n(\\h*\\$pipe)?([^\n]*)"),
+    Pattern.MULTILINE,
+  )
+
+  val stripMarginPatternWithLineContent =
+    compileStripMarginPatternWithLineContent('|')
+
+  // see: https://ammonite.io/#Save/LoadSession
+  @inline
+  val ammonitePattern: Regex = "(?:\\s*\\n@)".r
+
+  @inline
+  val stripMarginPattern = Pattern.compile(
+    RegexCompat.fixHorizontalSpaceInRegex("\n(\\h*\\|)?([^\n]*)"),
+    Pattern.MULTILINE,
+  )
+
+  // startAfterPattern and endBeforePattern should be unique in basePattern
+  // basePattern = startAfterPattern + matched pattern + endBeforePattern
+  private def replaceAll(
```

Comment:
Because a method like this is not part of the java stdlib (this is important for the way things are packaged there). Scala Native's regex implementation is unfortunately just a copy paste of re2j, converted into scala